File: frontend/src/types/stats.ts
export interface RecentExamen {
  study_id: string
  patient_id: string
  date: string
  vertebres: string[]
  score_global: number | null
  fracture_detectee: boolean | null
  analysed: boolean
}

export interface DashboardStats {
  today_exams: number
  month_fractures: number
  avg_score: number
  avg_time: number
  recent_exams: RecentExamen[]
}

export type StatsPeriod = '7d' | '30d' | '90d'

export interface DailyCount {
  date: string
  count: number
}

export interface ResultDistribution {
  label: string
  value: number
}

export interface VertebraStat {
  vertebre: string
  fracture_count: number
  avg_score: number
}

export interface RecallMetrics {
  recall: number
  auc: number
  f1: number
  model_name: string
}

export interface HistoriqueStats {
  period: StatsPeriod
  daily_counts: DailyCount[]
  result_distribution: ResultDistribution[]
  vertebrae_distribution: VertebraStat[]
  recall_metrics: RecallMetrics
}

File: backend/app/schemas/stats.py
from datetime import date, datetime

from pydantic import BaseModel, Field


class RecentExamenSchema(BaseModel):
    study_id: str
    patient_id: str
    date: datetime
    vertebres: list[str] = Field(default_factory=list)
    score_global: float | None = None
    fracture_detectee: bool | None = None
    analysed: bool = False


class DashboardStatsSchema(BaseModel):
    today_exams: int
    month_fractures: int
    avg_score: float
    avg_time: float
    recent_exams: list[RecentExamenSchema]


class DailyCountSchema(BaseModel):
    date: date
    count: int


class ResultDistributionSchema(BaseModel):
    label: str
    value: int


class VertebraStatSchema(BaseModel):
    vertebre: str
    fracture_count: int
    avg_score: float


class RecallMetricsSchema(BaseModel):
    recall: float
    auc: float
    f1: float
    model_name: str


class HistoriqueStatsSchema(BaseModel):
    period: str
    daily_counts: list[DailyCountSchema]
    result_distribution: list[ResultDistributionSchema]
    vertebrae_distribution: list[VertebraStatSchema]
    recall_metrics: RecallMetricsSchema
